### PR TITLE
Kornev/use rand gen

### PR DESCRIPTION
**Pull request by evgenykor**

### CODE REVIEW

**Comment by marcos-diaz** [comment]
ack, will be reviewed soon

**Review by przymi — APPROVED** [review]

File: src/common.c [hunk 1]
```diff
@@ -13,9 +13,8 @@ uint32_t bin16(uint16_t k) {
     return (k == 0 || k == 1 ? k : ((k % 2) + 10 * bin(k / 2)));
 }
 
-// Very basic randomization, do NOT use for anything serious.
 uint8_t random8() {
-    return (uint8_t)to_ms_since_boot(get_absolute_time());
+    return (uint8_t)get_rand_32();
```
Comment (marcos-diaz) [inline src/common.c:17]: There is possibly no need for this wrapper function anymore, the caller can use `get_rand_32()` directly.

File: src/logging.c [hunk 2]
```diff
@@ -70,6 +70,7 @@ void info(char *msg, ...) {
     va_list va;
     va_start(va, 0);
     write(msg, va);
+    va_end(va);
```
Comment (marcos-diaz) [inline src/logging.c]: Needs rebase/merge with main after the other PR was merged

File: src/common.c (hunk 1 above)
Comment (evgenykor) [reply src/common.c:17]: It is up to you of course but having a proper wrapper is better IMHO due to:

1) you anyway need 8 bit (not 32 bit of what pico provides) value so it makes sense to encapsulate it with explicit casting here.
2) if someday you will move out from pico you could just swap implementation of it with something else without touching actual logic

File: src/logging.c (hunk 2 above)
Comment (evgenykor) [reply src/logging.c]: addressed

**Pull request merged**